src/app/containers/RadioSchedule/Canonical/index.test.jsx

```diff
@@ -1,55 +1,153 @@
 import React from 'react';
+import fetchMock from 'fetch-mock';
 import { render, wait } from '@testing-library/react';
 import { matchSnapshotAsync } from '@bbc/psammead-test-helpers';
 import '@testing-library/jest-dom/extend-expect';
-import RadioSchedulesWithContext from '../utilities/testHelpers';
 import arabicRadioScheduleData from '#data/arabic/bbc_arabic_radio/schedule.json';
+import processRadioSchedule from '../utilities/processRadioSchedule';
+import CanonicalRadioSchedule from '.';
+import { RequestContextProvider } from '#contexts/RequestContext';
+import { ServiceContextProvider } from '#contexts/ServiceContext';
+
+const endpoint = 'https://localhost/arabic/bbc_arabic_radio/schedule.json';
+
+/* eslint-disable react/prop-types */
+const RadioScheduleWithContext = ({ initialData }) => (
+  <RequestContextProvider
+    isAmp={false}
+    pageType="frontPage"
+    service="arabic"
+    pathname="/arabic"
+    timeOnServer={Date.now()}
+  >
+    <ServiceContextProvider service="arabic">
+      <CanonicalRadioSchedule initialData={initialData} endpoint={endpoint} />
+    </ServiceContextProvider>
+  </RequestContextProvider>
+);
 
 describe('Canonical RadioSchedule', () => {
-  afterEach(() => {
-    fetch.resetMocks();
+  beforeEach(() => {
+    fetchMock.restore();
   });
 
-  it('renders correctly for a service with a radio schedule and page frequency URL', async () => {
-    fetch.mockResponse(JSON.stringify(arabicRadioScheduleData));
+  describe('With initial data', () => {
+    it('renders correctly for a service', async () => {
+      fetchMock.mock(endpoint, arabicRadioScheduleData);
+      const initialData = processRadioSchedule(
+        arabicRadioScheduleData,
+        'arabic',
+        Date.now(),
+      );
+      await matchSnapshotAsync(
+        <RadioScheduleWithContext initialData={initialData} />,
+      );
+      expect(fetchMock.calls(endpoint).length).toBeFalsy();
+    });
+
+    it('contains four programs for a service with a radio schedule', async () => {
+      const initialData = processRadioSchedule(
+        arabicRadioScheduleData,
+        'arabic',
+        Date.now(),
+      );
+      const { container } = render(
+        <RadioScheduleWithContext initialData={initialData} />,
+      );
 
-    await matchSnapshotAsync(
-      <RadioSchedulesWithContext service="arabic" radioScheduleToggle />,
-    );
-  });
+      await wait(() => {
```

Comment:
Looks good to me overall 💯
Wondering if we still have to replace this with `await act( async () => {`

Reply:
I would like to, I've added that to another ticket to do once this is reviewed, there's quite a lot of changes in here already!

Reply:
https://github.com/bbc/simorgh/issues/6219

Reply:
Excellent 👍